webapp/store/models.py

```diff
@@ -0,0 +1,145 @@
+        'id': int(revision_list[0].split('-')[-1]),
+        'full_id': revision_list[0],
+        'url': '{}/{}'.format(
+            meta['charm-metadata']['Name'],
+            int(revision_list[0].split('-')[-1])
+        )}
+
+    charm_or_bundle = {
+        'archive_url': cs.archive_url(ref),
+        'bugs_url': bugs_url,
+        'bzr_url': bzr_url,
+        'entity_data': entity_data,
+        'files': _get_entity_files(ref, meta['manifest']),
+        'homepage': homepage,
+        'icon': cs.charm_icon_url(charm_or_bundle_id),
+        'id': charm_or_bundle_id,
+        'card_id': ref.path(),
+        'latest_revision': latest_revision,
+        'options': meta.get('charm-config', {}).get('Options'),
+        'owner': entity_data.get('owner', {}).get('User'),
+        'provides': meta['charm-metadata'].get('Provides'),
+        'requires': meta['charm-metadata'].get('Requires'),
+        'resources': _extract_resources(ref, meta.get('resources', {})),
+        'readme': _render_markdown(
+            cs.entity_readme_content(charm_or_bundle_id)
+        ),
+        'revision_list': revision_list,
+        'revision_number': ref.revision,
+        'revisions': revisions,
+        'series': meta.get('supported-series', {}).get('SupportedSeries'),
+        'is_charm': 'charm-metadata' in meta
+    }
+
+    return charm_or_bundle
+
+
+def _get_entity_files(ref, manifest=None):
+    try:
+        files = cs.files(ref, manifest=manifest) or {}
+    except EntityNotFound:
+        files = {}
+    return files
+
+
+def _extract_resources(ref, resources):
+    """Extract data from resources metadata.
+    @param ref The reference of the entity.
+    @param resources the resources metadata associated with the entity.
+    @return a dictionary of resource name and an array containing
+            the file extension, the file link of the resource.
+    """
+    result = {}
+    for resource in resources:
+        resource_url = ""
+        if resource['Revision'] >= 0:
+            resource_url = cs.resource_url(
+                ref,
+                resource['Name'],
+                resource['Revision']
+            )
+        result[resource['Name']] = [
+            os.path.splitext(resource['Path'])[1],
+            resource_url
+        ]
+
+    return result
+
+
+def _get_bug_url(name, bugs_url):
+    '''Create a link to the bug tracker on Launchpad.
+    @param name the charm name.
+    @return a URL for the bug tracker.
+    '''
+    if bugs_url:
+        return bugs_url
+    return 'https://bugs.launchpad.net/charms/+source/{}'.format(name)
+
+
+def _render_markdown(content):
+    html = gfm.markdown(content)
+
+    try:
+        html = _convert_http_to_https(html)
+    except Exception:
+        # Leave the readme unparsed.
+        pass
+
+    return html
+
+
+def _convert_http_to_https(content):
+    '''Convert any non secure inclusion of assets to secure.
+        @param content the content to parse as a string.
```

Comment:
This is the correct syntax
```
        :param content: the content to parse as a string.
        :returns: the parsed content with http replaces with https
```

Reply:
@tbille which correct syntax? We don't currently use a docstring parser, and I believe different ones have different formats.

Reply:
what I'm saying is, I don't think we as a team currently follow a standard for docstring syntax.